### PR TITLE
[TASK] Add documentation for extbase newRecordStoragePid setting

### PR DESCRIPTION
**Pull request by derhansen**
Add documentation for settings.persistence.classes.[classname].newRecordStoragePid

### CODE REVIEW

File: Documentation/TopLevelObjects/Plugin.rst [hunk 1]
```diff
@@ -348,6 +359,10 @@ Demonstrates:
             storagePid = 42
             # Includes 4 sub-levels of the storagePid
             recursive = 4
+
+            Vendor\Extension\Domain\Model\MyModel {
```
Comment (mbrodala) [inline Documentation/TopLevelObjects/Plugin.rst:363]: This should be more specifically written for the blog example. Also it is unclear why one would need/set this without mentioning Single Table Inheritance. With only a single model for a single table, this option is not necessary. But if there are multiple models for a table with different record types, then this option is useful to store these on different pages.

**Review by mbrodala — CHANGES_REQUESTED** [review]

File: Documentation/TopLevelObjects/Plugin.rst (hunk 1 above)
Comment (derhansen) [reply Documentation/TopLevelObjects/Plugin.rst]: I agree with the blog example and have change the PR. 

The setting is however pretty self explaining and I would prefer to not reduce it to single table inheritance only (see see this https://typo3.slack.com/archives/C03AM9R17/p1716279888744799 discussion on slack).

Comment (mbrodala) [reply Documentation/TopLevelObjects/Plugin.rst]: Ah, that's correct indeed. This is not limited to SIT but actually about multiple models in an extension/plugin.

Can you still add another model for clarity?

Comment (derhansen) [reply Documentation/TopLevelObjects/Plugin.rst]: I already did, GitHub does not seem to update it in the commentes section. Change see: https://github.com/TYPO3-Documentation/TYPO3CMS-Reference-Typoscript/pull/1261/files#diff-2f734f75a0276d75461dfe447428e2d79a23b45b031da21e9071a3a8b9a02510R363

Comment (mbrodala) [reply Documentation/TopLevelObjects/Plugin.rst]: No, I saw that. I mean a 2nd model.

Comment (derhansen) [reply Documentation/TopLevelObjects/Plugin.rst]: Done

**Review by mbrodala — APPROVED** [review]

**Pull request merged**